recipe/meta.yaml

```diff
@@ -18,85 +18,160 @@ source:
-    - m2-patch  # [win]
   host:
-    - python
-    - libffi 3.4  # [not win]
-    # code is incompatible to newer libffi headers
-    - libffi 3.4   # [win]
-    - gettext   # [osx]
-    - zlib
-    - pcre
-    - libiconv  # [osx or win]
-  run:
-    - libffi >=3.4,<3.5
-    - gettext   # [osx]
-    - zlib
-    - pcre
-    - libiconv  # [osx or win]
+    - libffi {{ libffi }}
+    - gettext 0.21.0  # [osx]
+    - zlib {{ zlib }}
+    - pcre2 10.42
+    - libiconv 1.16
 
-test:
-  commands:
-    - test -f ${PREFIX}/lib/libglib-2.0{{ SHLIB_EXT }}  # [not win]
-    - test ! -f ${PREFIX}/lib/libgobject-2.0.la  # [not win]
-    # Check that binaries can run -- instigated by Meson Linux rpath issue
-    - gapplication help  # [linux]
-    - gdbus help
-    - gio version
-    - gio-querymodules .
-    - glib-compile-resources --help
-    - glib-compile-schemas --help
-    - gobject-query --help
-    - gresource help
-    - gtester --help  # [not win]
+outputs:
+  - name: libglib
+    script: install.sh  # [unix]
+    script: install.bat  # [not unix]
+    build:
+      ignore_run_exports_from:
+        - python *
+      run_exports:
+        - {{ pin_subpackage("libglib") }}
+    requirements:
+      build:
+        - meson >=0.60.0
+        - ninja
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - python >=3.5
+        - setuptools
+      host:
+        - libffi {{ libffi }}
+        - gettext 0.21.0  # [osx]
+        - zlib {{ zlib }}
+        - pcre2 10.42
+        - libiconv 1.16
+      run:
+        - gettext >=0.21.0,<1.0a0  # [osx]
+        # bounds through run_exports
+        - libffi
+        - zlib
+        - pcre2
+        - libiconv
+      run_constrained:
+        # Avoid colliding with older glib builds that didn't have a libglib output
+        - glib {{ version }} *_{{ PKG_BUILDNUM }}
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libglib-2.0.0{{ SHLIB_EXT }}  # [osx]
+        - test -f ${PREFIX}/lib/libglib-2.0{{ SHLIB_EXT }}.0  # [linux]
+        - if not exist "%LIBRARY_BIN%\\glib-2.0-0.dll" exit 1 # [win]
+        - if not exist "%LIBRARY_LIB%\\glib-2.0.lib" exit 1   # [win]
+        - test ! -f ${PREFIX}/lib/libgobject-2.0.la  # [not win]
+        - test ! -f ${PREFIX}/lib/libglib-2.0${SHLIB_EXT}  # [not win]
+        - test -f ${PREFIX}/lib/pkgconfig/glib-2.0.pc  # [unix]
+        - test -f ${PREFIX}/etc/conda/activate.d/libglib_activate.sh  # [not win]
+        - test -f ${PREFIX}/etc/conda/deactivate.d/libglib_deactivate.sh  # [not win]
+  - name: glib-tools
+    script: install.sh  # [unix]
+    script: install.bat  # [not unix]
+    build:
+      ignore_run_exports_from:
+        - python *
+    requirements:
+      build:
+        - meson >=0.60.0
+        - ninja
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - python >=3.5
+        - setuptools
+        - gettext 0.21.0  # [osx]
+      host:
+        - {{ pin_subpackage("libglib", exact=True) }}
```

Comment:
zlib not required? it got added in c-f

Reply:
From what I saw, ony libglib has binaries linking to zlib.